nsq/writer.py

```diff
@@ -143,8 +147,9 @@ def _pub(self, command, topic, msg, delay_ms=None, callback=None):
 
         try:
             conn.send(cmd(*args))
-        except Exception:
+        except:
```

Comment:
Why this change?

It's generally a good idea to catch no more broadly than `Exception`. There are a couple exceptions which intentionally do not subclass it, like `SystemExit` and `KeyboardInterrupt`, because catching those can (intermittently) break exiting and restarting the process.